### PR TITLE
How to Create a machine learning model using Turi Create

### LINKED ISSUE #2947: How to Create a machine learning model using Turi Create
### Proposed title of article
How to Create a machine learning model using Turi Create 

### Introduction paragraph (2-3 paragraphs):
Turi Create is an open-source toolset for creating Core ML models, for tasks such as image classification, object detection, style transfers, recommendations. Turi Create simplifies the development of custom machine learning models. Turi Create was made by Apple and is a powerful tool for machine learning.
In this tutorial, we will use Turi Create to show us how to read CSV files, how to create data frames, and basic data manipulation. Eventually, we will create a machine learning model using this package, this will teach us a lot about the Turi Create library.
Benefits of using Turi create:
  - Easy-to-use: Focus on tasks instead of algorithms
 - Visual: Built-in, streaming visualizations to explore your data
 - Flexible: Supports text, images, audio, video, and sensor data
 - Fast and Scalable: Work with large datasets on a single machine 
### Key takeaways: 
 - How to install Turi Create into our machines.
 - How to create data frames.
 - Data manipulation using turi create.
 - Data Preprocessing using turi create.
 - Building a Classification Model.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by willyngashu**
This fixes #2947

### Article checklist

**ATTENTION: In your PR - add the ISSUE (topic form) # that this PR corresponds to. You can do this by also adding a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.**
      
      
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**If you have not already please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve our overall technical writing.**

#### Preliminary Checks - Formatting and Structure 
- [ ] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [ ] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [ ] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [ ] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [ ] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [ ] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [ ] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [ ] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [ ] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [ ] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [ ] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [ ] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Good afternoon and thank you for submitting your article for publishing.
Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. :rocket:
Thank you. @willyngashu

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+ str,
+ str,
+ str,
+ str,
+ str,
+ str,
+ str]
+```
+
+Our dataset is made up of strings and integers.
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:140]: Specify that the first column is an integer and the remaining are strings

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+ 'Irritability',
+ 'delayed healing',
+ 'partial paresis',
+ 'muscle stiffness',
+ 'Alopecia',
+ 'Obesity',
+ 'class']
+```
+
+Our target will be the class column.
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:180]: The column titled `class` is our target variable.

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+
+### Getting Features
+
+```python
+feature_names = ['Age',
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:185]: Explain what you did here. Explain all code snippets in detail.

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+In this phase, we will start building our model using the given dataset above.
+Before we begin, we need to split our dataset into a training set and a testing set. The training set will be 75%, and the testing set will be 25%.
+
+### Dataset splitting
+
+```python
+train_data,test_data = df.random_split(0.75)
+```
+
+- Shape of the original data.
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:214]: Use H4 for such headings.

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+
+```bash
+(390, 17)
+```
+
+After splitting our data into a training and a testing set, we can begin building our model.
+
+### Modelling Algorithm
+
+TuriCreate supports different classification algorithms, in our case, we will use the [Logistic regression](https://en.wikipedia.org/wiki/Logistic_regression)
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:242]: Explain why logistic regression is a good choice in a line or two.

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+
+- To use the logistic regression algorithm use the foloowing command:
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:244]: following

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+| 2         | 3        | 0.016456    | 0.931734         | 0.843725            |
+| 3         | 4        | 0.018732    | 0.949024         | 0.843725            |
+| 4         | 5        | 0.020935    | 0.957320         | 0.843725            |
+| 5         | 6        | 0.023619    | 0.960135         | 0.843725            |
+| 7         | 8        | 0.026907    | 0.969716         | 0.843725            |
++-----------+----------+--------------+-------------------+---------------------+
+SUCCESS: Optimal solution found.
+```
+
+TuriCreate will be able to use both the training accuracy and Validation accuracy through the iteration. Here we iterate 7 times, and the Training accuracy and Validation accuracy after 7 iterations will be 0.965517 0.842105, respectively.
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:278]: What do you mean by the first sentence? It is not clear.

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+
+### Getting the model summary
+
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:281]: Give an outline of what to expect from each section/ sub-section. If we jump into the code directly, we lose the interest of the reader.

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 9]
```diff
@@ -0,0 +1,413 @@
...
+----------------------------
+Polydipsia[No]                 : -6.5078
+Alopecia[No]                   : -0.8826
+weakness[No]                   : -0.283
+Age                            : -0.0729
+```
+
+### Model Evaluation
+
+This is assessing our model to find out how well it learned. We do this by using the test_data
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:333]: Whenever you refer to variables used in code, use \`\` to wrap the variable. Example: \`test_data\`

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 10]
```diff
@@ -0,0 +1,413 @@
...
+metrics
+```
+
+```python
+metrics['accuracy']
+```
+
+### Making Predictions
+
+- Rule of making a prediction
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:351]: What is an SFrame? Explain it in detail

File: content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md [hunk 11]
```diff
@@ -0,0 +1,413 @@
...
+
+Use the following command to save our model.
+
+```python
+logistic_model.save('diabetes_prediction.model')
+```
+
+### Conclusion
+
+In this tutorial, we have learned how to create a machine learning model using Turi Create. We started by creating SFrames to load our dataset. We then did data manipulation for us to know the structure of the data we are working with. Finally, after properly understanding our data, we started to build a machine learning model to predict if a person is at risk of getting diabetes.
```
Comment (lalith1403) [inline content/articles/how-to-create-a-machine-learning-app-using-turicreate/index.md:407]: Break this down into smaller paragraphs

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Hey @willyngashu, Thanks for submitting an interesting article. Look into the changes requested and we can move onto the next round of review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by willyngashu** [comment]
Hello, @lalith1403 I've made the requested changes, kindly review.
Thanks.

**Comment by willyngashu** [comment]
Hello @lalith1403, what is the progress for this article?

**Comment by ahmadmardeni1** [comment]
I'm always using Sklearn. I didn't use TuriCreate before, but definitely, I will! It is very easy to follow.
Great article @willyngashu

Thank you for the review @lalith1403
